packages/plugin-js-packages/src/lib/runner/outdated/transform.ts

```diff
@@ -0,0 +1,123 @@
+import { Issue } from '@code-pushup/models';
+import { objectToEntries } from '@code-pushup/utils';
+import { DependencyGroup } from '../../config';
+import { outdatedSeverity } from './constants';
+import {
+  NormalizedOutdatedEntries,
+  NormalizedVersionOverview,
+  NpmOutdatedResultJson,
+  PackageVersion,
+  VersionType,
+} from './types';
+
+export function outdatedResultToAuditOutput(
+  result: NpmOutdatedResultJson,
+  dependenciesType: DependencyGroup,
+) {
+  // current might be missing in some cases
+  // https://stackoverflow.com/questions/42267101/npm-outdated-command-shows-missing-in-current-version
+  const validDependencies: NormalizedOutdatedEntries = objectToEntries(result)
+    .filter(
+      (entry): entry is [string, NormalizedVersionOverview] =>
+        entry[1].current != null,
+    )
+    .filter(([, detail]) =>
+      dependenciesType === 'prod'
+        ? detail.type === 'dependencies'
+        : detail.type === `${dependenciesType}Dependencies`,
+    );
+  const outdatedDependencies = validDependencies.filter(
+    ([, versions]) => versions.current !== versions.wanted,
+  );
+
+  const majorOutdatedAmount = outdatedDependencies.filter(
+    ([, versions]) =>
+      getOutdatedLevel(versions.current, versions.wanted) === 'major',
+  ).length;
+
+  const issues =
+    outdatedDependencies.length === 0
+      ? []
+      : outdatedToIssues(outdatedDependencies);
+
+  return {
+    slug: `npm-outdated-${dependenciesType}`,
+    score: calculateOutdatedScore(
+      majorOutdatedAmount,
+      validDependencies.length,
+    ),
+    value: outdatedDependencies.length,
+    displayValue: outdatedToDisplayValue(
+      majorOutdatedAmount,
+      outdatedDependencies.length,
+    ),
+    ...(issues.length > 0 && { details: { issues } }),
+  };
+}
+
+export function calculateOutdatedScore(
+  majorOutdated: number,
+  totalDeps: number,
+) {
+  return totalDeps > 0 ? (totalDeps - majorOutdated) / totalDeps : 1;
+}
+
+export function outdatedToDisplayValue(
+  majorOutdated: number,
+  totalOutdated: number,
+) {
+  return totalOutdated === 0
+    ? 'all dependencies are up to date'
+    : majorOutdated > 0
+    ? `${majorOutdated} out of ${totalOutdated} outdated dependencies require major update`
+    : `${totalOutdated} outdated ${
+        totalOutdated === 1 ? 'dependency' : 'dependencies'
+      }`;
```

Comment:
Perhaps we could format it similarly to auditted packages, so that the levels are always listed :thinking: Something like `6 outdated package versions (1 major, 2 minor, 3 patch)`.

But what you did is also good :slightly_smiling_face:

Reply:
Let us do that in another PR along with the cross references, this would require a refactor.